Add GET requests and deletion of Old Images

marl0vv opened the pull request:
Добавлен GET /convert-to-webp/async/{UUID}/status для получения статуса выполнения запроса.
Добавлен GET /convert-to-webp/async/{UUID} для получения конвертированного изображения.
Добавлен метод deleteOldConvertedImages для удаления старых изображений по истечении количества времени, указанного в application.properties

marl0vv committed:
Add GET request for status

Add GET request that returns status. If status is "ERROR" then also returns errorMessage

marl0vv committed:
Add GET request for converted image

Add GET request that returns a converted image or 404 if image is not ready of there was an error in conversion

marl0vv committed:
Add deletion of old converted images after some time

Add deletion of old converted images after time that is contained in application.properties

RavenVSS commented on src/main/java/com/glamik/webpconverter/controller/ConverterAsyncController.java:
```diff
@@ -25,4 +35,35 @@ public ResponseEntity<UUID> convertImageAsync(@RequestParam("image") MultipartFi
+        ConversionTask conversionTask = conversionTaskService.getConversionTask(taskId);
+        ConversionTaskStatus status = conversionTask.getStatus();
+
+        ConversionTaskStatusResponse response = ConversionTaskStatusResponse.builder()
```
https://javarush.com/groups/posts/3139-spring---ehto-ne-strashno-prosloyka-iz-dto

RavenVSS commented on src/main/java/com/glamik/webpconverter/controller/ConverterAsyncController.java:
```diff
@@ -25,4 +35,35 @@ public ResponseEntity<UUID> convertImageAsync(@RequestParam("image") MultipartFi
     }
 
+    @GetMapping("/convert-to-webp/async/{taskId}/status")
+    public ResponseEntity<ConversionTaskStatusResponse> getTaskStatus(@PathVariable UUID taskId) {
```
Можно возвращать сам объект DTO

RavenVSS commented on src/main/java/com/glamik/webpconverter/controller/ConverterAsyncController.java:
```diff
@@ -25,4 +35,35 @@ public ResponseEntity<UUID> convertImageAsync(@RequestParam("image") MultipartFi
+        ConversionTaskStatusResponse response = ConversionTaskStatusResponse.builder()
+                .status(status)
+                .errorMessage(status == ConversionTaskStatus.ERROR ? conversionTask.getErrorMessage() : null)
+                .build();
```
Попробуй подключить mapstruct

RavenVSS commented on src/main/java/com/glamik/webpconverter/scheduler/ConversionTaskScheduler.java:
```diff
@@ -26,4 +35,17 @@ public void processPendingConversionTasks() {
 
+
+    @Scheduled(initialDelayString = "${deletion.time.millis}", fixedDelayString = "${deletion.time.millis}")
+    public void deleteOldConvertedImages() {
```
Вынести в отдельный класс

RavenVSS commented on src/main/java/com/glamik/webpconverter/scheduler/ConversionTaskScheduler.java:
```diff
@@ -26,4 +35,17 @@ public void processPendingConversionTasks() {
+        List<ConversionTask> pendingTasks = conversionTaskRepository.findByStatusOrderByTaskCreationDate(ConversionTaskStatus.SUCCESS);
+
+        for (ConversionTask task : pendingTasks) {
+            long timeBetween = Duration.between(task.getTaskCreationDate(), LocalDateTime.now()).toMillis();
```
Нужно хранить время обработки в базе и уже от этого времени считать срок хранения файла

RavenVSS commented on src/main/java/com/glamik/webpconverter/scheduler/ConversionTaskScheduler.java:
```diff
@@ -26,4 +35,17 @@ public void processPendingConversionTasks() {
+
+        for (ConversionTask task : pendingTasks) {
+            long timeBetween = Duration.between(task.getTaskCreationDate(), LocalDateTime.now()).toMillis();
+            if (timeBetween >= deletionTimeMillis) {
```
Тянуть из базы сразу нужные файлы для удаления  

Для этого можно использовать аннотацию `@Query` в репозитории для создания запросов Sql

marl0vv committed:
Include MapStruct to project

marl0vv committed:
Use MapStruct for mapping dto

marl0vv committed:
Add processing date to model

Add processing date to model that allows us for deletion of files after a correct interval

marl0vv committed:
Get files for deletion directly from database

RavenVSS commented on src/main/java/com/glamik/webpconverter/service/ConversionTaskService.java:
```diff
@@ -47,4 +54,18 @@ public void setConversionErrorStatus(UUID id, ConversionTaskStatus status, Error
     }
 
+
+    public ConversionTask getConversionTask(UUID id) {
```
`@Transactional(readOnly=true)`

marl0vv committed:
Fix minor things
